testsuite/src/it/java/org/graalvm/tests/integration/AppReproducersTest.java

```diff
@@ -512,108 +538,131 @@ public void imageioAWTTest(TestInfo testInfo) throws IOException, InterruptedExc
+                "mytest.wbmp",
+                "mytest_Resized_Grace_M._Hopper.png"
+        );
+
         try {
-            // Cleanup
+            // Pre-run cleanup, just in case.
+            pictures.forEach(f -> new File(appDir, f).delete());
             cleanTarget(app);
             if (metaINF.exists()) {
                 FileUtils.cleanDirectory(metaINF);
             }
+            if (inContainer) {
+                for (String base : RUNTIME_IMAGE_BASE) {
+                    removeContainer(app.runtimeContainer.name + "_" + base);
+                }
+            }
+
+            // Logs dir
             Files.createDirectories(Paths.get(appDir.getAbsolutePath() + File.separator + "logs"));
 
             // Build
             processLog = Path.of(appDir.getAbsolutePath(), "logs", "build-and-run.log").toFile();
-
             builderRoutine(app, report, cn, mn, appDir, processLog, null, getSwitches(app));
 
-            // Record images' hashsums as created by a Java process
+            // Record images' pixels as created by a Java process, HotSpot mode
             final List<String> errors = new ArrayList<>(12);
-            List<String> pictures = List.of(
-                    "mytest.bmp",
-                    "mytest.gif",
-                    "mytest.jpg",
-                    "mytest.png",
-                    "mytest.svg",
-                    "mytest.tiff",
-                    "mytest_toC.png",
-                    "mytest_toG.png",
-                    "mytest_toL.png",
-                    "mytest_toP.png",
-                    "mytest_toS.png",
-                    "mytest.wbmp",
-                    "mytest_Resized_Grace_M._Hopper.png"
-            );
-
             pictures.forEach(f -> {
+                final File picture = new File(appDir, f);
                 try {
-                    controlData.put(f, DigestUtils.sha256Hex(FileUtils.readFileToByteArray(new File(appDir, f))));
-                } catch (IOException e) {
-                    errors.add(f + " cannot be loaded.");
+                    final BufferedImage image = f.endsWith("svg") ? ImageIO.read(new ByteArrayInputStream(
+                            Base64.getDecoder().decode(getSubstringFromSmallTextFile(svgWrapperPattern, picture.toPath(), UTF_8)))) :
+                            ImageIO.read(picture);
+                    final Map<Integer[], Integer[]> pixelXYValueRGBA = new HashMap<>();
+                    for (Integer[] xy : new Integer[][] {
+                            { 30, 51 },
+                            { 250, 250 },
+                            { 374, 374 } }) {
+                        final int[] rgba = new int[4]; //4BYTE RGBA, A optional
+                        image.getData().getPixel(xy[0], xy[1], rgba);
+                        pixelXYValueRGBA.put(xy, new Integer[] { rgba[0], rgba[1], rgba[2], rgba[3] });
+                    }
+                    imageNamePixelXYValueRGBA.put(f, pixelXYValueRGBA);
```

Comment:
If I get this right, you are changing the way this test works by comparing the RGBA value of 3 pixels while previously the test would compare the sha256 value of the whole image.

Why? This seems unrelated to the linux-image testing part. I suggest splitting these changes in a separate commit.

Reply:
@zakkak It's a long overdue change that I had implemented in Quarkus AWT a long time ago: 

https://github.com/quarkusio/quarkus/blob/main/integration-tests/awt/src/test/java/io/quarkus/awt/it/ImageEncodersTest.java#L85

Note that this change *does not relax* the test, it tightens it for containers. Due to the differences in fontconfig, freetype and previously out-of-tree harfbuzz, the testing of hashsums was skipped for running in container and the test had been merely checking file sizes.

On the other hand, for the locally installed and locally executed test tun, without containers, this *relaxes* the test, from hashsum to mere pixels check.

Reply:
Thanks for the additional info. Would it make sense to have a different test for containers then to get the best of both worlds? (just asking, I have no strong opinion)

In any case, please consider splitting the change in a separate commit.

Reply:
Ack, I will:
 * keep the hashsum check for local run
 * leave this new check for containers

The overall reason of my focus here is not any particular fondness of AWT, it's just the easiest way to verify various  dynamically loaded libs work, it has by far the most dependencies while being easily portable across Linux distros.

So if AWT fails, we know symbols/linkage/something is broken...